src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,229 @@
+
+# Example
+
+```jldoctest
+julia> S4 = quantum_symmetric_group(4);
+
+julia> length(gens(S4))
+120
+```
+"""
+function quantum_symmetric_group(n::Int)
+  A, u = free_associative_algebra(QQ, :u => (1:n, 1:n))
+
+  relations = elem_type(A)[]
+
+  #Idempotent relations
+  for i in 1:n, j in 1:n
+    new_relation = u[i,j] * u[i,j] - u[i,j]
+    push!(relations, new_relation)
+    for k in 1:n
+      if k != j
+        new_relation = u[i,j] * u[i,k]
+        push!(relations, new_relation)
+        new_relation = u[j,i] * u[k,i]
+        push!(relations, new_relation)
+      end
+    end
+  end
+
+  #row and column sum relations
+  for i in 1:n
+    push!(relations, sum(u[i,:]) - 1)
+    push!(relations, sum(u[:,i]) - 1)
+
+  end
+  return ideal(relations)
+end
+
+@doc raw"""
+    _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+
+Return the indices of the relations that define the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+julia> M = uniform_matroid(3,4);
+
+julia> idx = Oscar._quantum_automorphism_group_indices(M,:bases);
+
+julia> length(idx)
+1920
+```
+"""
+function _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+  @req structure in [:bases, :circuits, :flats] "structure must be one of :bases, :circuits, :flats"
+
+  matroid_groundset(M) isa Vector{Int} ? nothing : M=Matroid(Oscar.pm_object(M))  
+
+  n = length(M)
+  grdSet = matroid_groundset(M)
+
+  rels = Vector{Tuple{Int,Int}}[]
+
+  sets  = getproperty(Oscar, structure)(M)
+  sizes = unique!(map(length, sets))
+
+  function _qAut_size()
+    setSizes = map(length, sets)
+
+    SetsCount = [0 for _ in 1:maximum(setSizes)]
+    foreach(size->SetsCount[size] += 1, setSizes)
+    for i in 1:length(SetsCount)
+      if SetsCount[i] == 0 || SetsCount[i] == n^i
+        SetsCount[i] = 0
+        continue
+      end
+      nonSetsCount = n^i - (SetsCount[i] * factorial(i))
+      SetsCount[i] = SetsCount[i] * factorial(i) * nonSetsCount * 2
+    end
+
+    return sum(SetsCount)
+  end
+  vector_size = _qAut_size()
+  sizehint!(rels, vector_size)
+
+  for size in sizes 
+    size == 0 && continue
+
+    S = symmetric_group(size)
+    permutedSets = [[set[g(i)] for i in 1:size] for g in S for set in sets if length(set) == size]
+
+    nonSets = MultiPartitionIterator{Int}(grdSet, size; toskip=permutedSets)
+
+    for nonset in nonSets
+      for set in permutedSets
+        push!(rels, [(set[i],nonset[i]) for i in 1:size])
+        push!(rels, [(nonset[i],set[i]) for i in 1:size])
+      end
+    end
+  end
+  @assert length(rels) == vector_size "The number of relations is not correct $(length(rels)) != $(vector_size)"
+
+  return Vector{Vector{Tuple{Int,Int}}}(rels)
+end
+
+@doc raw"""
+    quantum_automorphism_group(M::Matroid, structure::Symbol=:bases)
+
+Return the Ideal that defines the quantum automorphism group of a matroid for a given structure.
```

Comment:
I  have a feeling of deja vu ;-)
```suggestion
Return the ideal that defines the quantum automorphism group of a matroid for a given structure.
```

Reply:
Yes, I wasn't able to commit to your previous suggestion :) sooorry.